src/main/java/org/opensearch/neuralsearch/search/query/TopDocsMerger.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import org.apache.lucene.search.FieldDoc;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TopFieldDocs;
+import org.apache.lucene.search.TotalHits;
+import org.opensearch.common.lucene.search.TopDocsAndMaxScore;
+
+import java.util.Comparator;
+import java.util.Objects;
+import org.opensearch.search.sort.SortAndFormats;
+
+/**
+ * Utility class for merging TopDocs and MaxScore across multiple search queries
+ */
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+class TopDocsMerger {
+
+    private HybridQueryScoreDocsMerger<ScoreDoc> scoreDocsMerger;
+    private HybridQueryScoreDocsMerger<FieldDoc> fieldDocsMerger;
```

Comment:
why do we need two merger instances if we're initializing and use them in exactly same way? Can we do same with only one `docsMerger`?

Reply:
Because HybridQueryScoreDocsMerger has 
```
class HybridQueryScoreDocsMerger<T extends ScoreDoc>
```
class definition. Now if we have to operate on FieldDoc then we have to initialize score doc merger object with FieldDoc. If I use the same scoreDocsMergerObject for both then

```
public T[] merge(final T[] sourceScoreDocs, final T[] newScoreDocs, final Comparator<T> comparator, final boolean isSortEnabled)
```
comparator<T> comparator throws error.

Reply:
you can declare type of merger in raw format `HybridQueryScoreDocsMerger` and init with `new HybridQueryScoreDocsMerger<>()` or `new HybridQueryScoreDocsMerger<FieldDoc>()` depending on scenario (sorting/no sorting). 
Method `getMergedScoreDocs` can also be simplified, it can be something like `return merger.merge(source, newScoreDocs, comparator(), isSortingEnabled())` without if/else condition